### DISCUSSION TITLE
Is it possible to make local yggdrasil instance "memorize" the paths of a service hosted on its network.

### DISCUSSION THREAD

**Discussion by przemyslaw0** [yggdrasil-network/yggdrasil-go#1061] [Q&A] [answered]
When I query a domain using a DNS resolver of yggdrasil network, the consecutive queries are faster:

```
# time dig @324:71e:281a:9ed3::53 +short cloudflare.com
104.16.132.229
104.16.133.229

real	0m1.357s
user	0m0.007s
sys	0m0.002s

# time dig @324:71e:281a:9ed3::53 +short cloudflare.com
104.16.133.229
104.16.132.229

real	0m0.238s
user	0m0.007s
sys	0m0.002s

```

Is it possible to make my local yggdrasil instance memorize the path of a server I connected before for faster access afterwards?

If it is, how can I configure my yggdrasil to do that?

**Comment by majestrate** [comment]
this reduction in query latency is likely from your local dns cache, not yggdrasil's path latency.

**Comment by neilalexander** [comment, marked as answer]
It is not possible to pin the path to a destination node in the way you are suggesting, as that path could be invalidated at any time by the network topology changing between your node and the remote side.

The initial latency is down to Yggdrasil setting up the path and exchanging encryption keys with the remote side to create a session. That session will usually stay alive as long as traffic continues to be exchanged, but if the session is torn down from inactivity or connection interruption, Yggdrasil will have to tear down that session and establish a new one.
